test/ca/mcgill/cs/stg/jetuml/commands/TestComponudCommand.java

```diff
@@ -0,0 +1,272 @@
+import ca.mcgill.cs.stg.jetuml.graph.CallEdge;
+import ca.mcgill.cs.stg.jetuml.graph.ClassNode;
+import ca.mcgill.cs.stg.jetuml.graph.Edge;
+import ca.mcgill.cs.stg.jetuml.graph.Graph;
+import ca.mcgill.cs.stg.jetuml.graph.Node;
+
+public class TestComponudCommand 
+{
+    private Graph aGraph;
+    private Node aClassNode;
+    private Node aActorNode;
+    private Edge aCallEdge;
+    private Field aNodesToBeRemoved;
+    private Field aEdgesToBeRemoved;
+    private Field aNeedsLayout;
+    private Field aRootNodes;
+    private AddNodeCommand aAddClassNodeCommand;
+    private AddNodeCommand aAddActorNodeCommand;
+    private AddEdgeCommand aAddEdgeCommand;
+    private MoveCommand aMoveCommand1;
+    private MoveCommand aMoveCommand2;
+    private DeleteNodeCommand aDeletNodeCommand;
+    private CompoundCommand aCompoundCommand1;
+    private CompoundCommand aCompoundCommand2;
+    private CompoundCommand aCompoundCommand3;
+    private CompoundCommand aCompoundCommand4;
+    
+    @Before
+    public void setUp() throws Exception
+    {
+        aGraph = new ClassDiagramGraph();
+        aClassNode = new ClassNode();
+        aActorNode = new ActorNode();
+        aCallEdge = new CallEdge();
+        
+        aAddClassNodeCommand = new AddNodeCommand(aGraph, aClassNode);
+        aAddEdgeCommand = new AddEdgeCommand(aGraph, aCallEdge);
+        aAddActorNodeCommand = new AddNodeCommand(aGraph, aActorNode);
+        aMoveCommand1 = new MoveCommand(aGraph, aClassNode, 5, 5);
+        aMoveCommand2 = new MoveCommand(aGraph, aClassNode, -3, 3);
+        aDeletNodeCommand = new DeleteNodeCommand(aGraph, aClassNode);
+        
+        aNodesToBeRemoved = aGraph.getClass().getSuperclass().getDeclaredField("aNodesToBeRemoved");
+        aNodesToBeRemoved.setAccessible(true);
+        aEdgesToBeRemoved = aGraph.getClass().getSuperclass().getDeclaredField("aEdgesToBeRemoved");
+        aEdgesToBeRemoved.setAccessible(true);
+        aNeedsLayout = aGraph.getClass().getSuperclass().getDeclaredField("aNeedsLayout");
+        aNeedsLayout.setAccessible(true);
+        aRootNodes = aGraph.getClass().getSuperclass().getDeclaredField("aRootNodes");
+        aRootNodes.setAccessible(true);
+        
+        aCompoundCommand1 = new CompoundCommand();
+        aCompoundCommand1.add(aAddClassNodeCommand);
+        aCompoundCommand1.add(aAddEdgeCommand);
+        aCompoundCommand1.add(aMoveCommand1);
+        
+        aCompoundCommand2 = new CompoundCommand();
+        aCompoundCommand2.add(aAddActorNodeCommand);
+        aCompoundCommand2.add(aMoveCommand2);
+        
+        aCompoundCommand3 = new CompoundCommand();
+        aCompoundCommand3.add(aCompoundCommand1);
+        aCompoundCommand3.add(aCompoundCommand2);
+        
+        aCompoundCommand4 = new CompoundCommand();
+        aCompoundCommand4.add(aCompoundCommand1);
+        aCompoundCommand4.add(aMoveCommand2);
+        aCompoundCommand4.add(aDeletNodeCommand);
+    }
+
```

Comment:
Please don't use underscores in method names. You can just do the camelCase version.